docs: document defaultFrameworks for operator installs (#135)
kubescape/kubescape.io: docs/docs/install-operator.md

You can configure these by using `--set` when installing the chart, or by specifying your own values file with the `-f` flag. [Read the Helm documentation to learn more](https://helm.sh/docs/chart_template_guide/values_files/).

> ### Default posture frameworks

Reviewer: **Blocker — documents an unreleased value.**

`defaultFrameworks` is not in the released chart. It comes from [helm-charts#890](https://github.com/kubescape/helm-charts/pull/890) (open) and is only honored by the operator via [operator#398](https://github.com/kubescape/operator/pull/398) (open). Neither is merged, and `helm-charts@main`'s `values.yaml` has no such key.

Because the site is unversioned and the chart has no values schema, a reader following this today gets a Helm command that succeeds and does nothing.

Gate the merge on both PRs landing in a released chart, and add a minimum-version line here:

```
!!! note
    Requires kubescape-operator chart vX.Y.Z or later.
```

> Choose which security frameworks the operator uses when a scan does not specify frameworks explicitly (startup scan, scheduled CronJob with empty `scanV1`, and exception-triggered rescans):

Reviewer: Worth being explicit that this list is only consulted when the request carries **no** `targetNames`. In `operator#398`, `setDefaultsKubescapeScanRequest` applies `defaultFrameworks` under `if len(postScanRequest.TargetNames) == 0` — the target *type* doesn't matter, only whether names were supplied.

```yaml
defaultFrameworks:
  - allcontrols
  - nsa
  - mitre
```

Examples:

```shell

> # AKS CIS only

Reviewer: **Blocker — "AKS CIS only" is not what runs.**

The operator appends the `security` framework to every framework-type scan when `kubescape.triggerSecurityFramework` is true, which is the chart default (`values.yaml:421`):

```go
// mainhandler/kubescapehandler.go:170-173
if actionHandler.config.TriggerSecurityFramework() {
    appendSecurityFramework(request)
}
```

So this install scans `cis-aks-t1.2.0` **and** `security`. Users will see security-framework results they were told not to expect.

Either drop "only" (`# AKS CIS benchmark`) or add a sentence after the examples noting that `security` is also scanned unless `kubescape.triggerSecurityFramework=false`.

helm upgrade --install kubescape kubescape/kubescape-operator -n kubescape --create-namespace \
  --set clusterName=`kubectl config current-context` \
  --set defaultFrameworks={cis-aks-t1.2.0}

> # GKE-oriented set

Reviewer: **Blocker — this is not a GKE framework.**

`cis-v1.23-t1.0.1` is the CIS Kubernetes Benchmark **v1.23 for vanilla Kubernetes**, per this very docs site (`frameworks-and-controls/frameworks.md`):

> **CIS Kubernetes Benchmark v1.23** (`cis-v1.23-t1.0.1`) - For Kubernetes version 1.23

Kubescape ships no GKE CIS benchmark — the set is `cis-v1.10.0`, `cis-v1.23-t1.0.1`, `cis-eks-t1.7.0`, `cis-aks-t1.2.0`. Labelling it "GKE-oriented" is wrong, and it also steers readers to a benchmark for Kubernetes 1.23 when `cis-v1.10.0` is the current one (k8s 1.30/1.31).

Suggest renaming the example to something honest, e.g.:

```shell
# All controls plus the current CIS Kubernetes benchmark
... --set 'defaultFrameworks={allcontrols,cis-v1.10.0}'
```

The same wrong comment is in `values.yaml` in helm-charts#890 — worth fixing there too so they don't diverge.

helm upgrade --install kubescape kubescape/kubescape-operator -n kubescape --create-namespace \
  --set clusterName=`kubectl config current-context` \
  --set 'defaultFrameworks={allcontrols,cis-v1.23-t1.0.1}'
```

> API-triggered scans that pass `targetNames` still override this list. To override frameworks for the scheduled CronJob only, see [Scheduled scans](operator/scheduled-scans.md).

Reviewer: Agreed with CodeRabbit that this needs scoping, though its suggested wording is slightly off in the other direction. The operator doesn't check the target *type* when deciding whether to apply the defaults — `setDefaultsKubescapeScanRequest` only fires when `targetNames` is empty. So *any* explicit `targetNames`, framework or control, suppresses `defaultFrameworks`.

More accurate:

```suggestion
Any scan that explicitly sets `targetNames` — including API-triggered scans — uses those targets instead of this list. To override frameworks for the scheduled CronJob only, see [Scheduled scans](operator/scheduled-scans.md).
```

Minor, while you're here: the two `--set` examples quote inconsistently (`{cis-aks-t1.2.0}` bare vs `'{allcontrols,...}'`). Quote both — it's copy-paste-safe across shells.

### Configuring parameters

> See [the GitHub repository for the Kubescape operator](https://github.com/kubescape/helm-charts/blob/main/charts/kubescape-operator/README.md#chart-support) to learn the full set of configuration parameters.

Reviewer: This sends readers to the chart README for "the full set of configuration parameters", but helm-charts#890 doesn't add `defaultFrameworks` to that README's Values table. Net result: the parameter is documented here and missing from the reference this page points at. Worth adding the row in #890 before this merges.

You can configure these by using `--set` when installing the chart, or by specifying your own values file with the `-f` flag. [Read the Helm documentation to learn more](https://helm.sh/docs/chart_template_guide/values_files/).

### Default posture frameworks

!!! note

>     Requires a kubescape-operator chart release that includes `defaultFrameworks` (and an operator image that reads it). On older charts the Helm `--set` is accepted but ignored.

Reviewer: The note is the right shape, but it can't be acted on yet: helm-charts#890 and operator#398 are both still open, so **no** published chart includes `defaultFrameworks`, and "a chart release that includes it" gives the reader nothing to check against.

Suggest holding the merge until both land and a chart release ships, then making this concrete:

```
!!! note
    Requires kubescape-operator chart vX.Y.Z or later (and the operator image it pins). On older charts the Helm `--set` is accepted but ignored.
```

>     Requires a kubescape-operator chart release that includes `defaultFrameworks` (and an operator image that reads it). On older charts the Helm `--set` is accepted but ignored.

Reviewer: This is the one remaining blocker, and it's now a single edit away.

`Chart.yaml` on helm-charts `main` is still `1.40.3`, and that tag was published 2026-07-22 — before #890 merged. No published chart carries `defaultFrameworks`, so today this note asks the reader to check for something they cannot get, and the `--set` is still a silent no-op.

Once the chart release is cut, make it checkable:

```
!!! note
    Requires kubescape-operator chart vX.Y.Z or later (which pins operator v0.2.161 or newer). On older charts the Helm `--set` is accepted but ignored.
```

The operator half is already done — v0.2.161 shipped with operator#398 and chart `main` pins it. Only the chart version bump + release is missing.

> Choose which security frameworks the operator uses when a scan request has **no** `targetNames` (empty scheduled `scanV1`, API scans without targets, startup scan, and exception-triggered rescans).

Reviewer: Nit — worth adding continuous scanning to this list. operator#398's own `docs/features/default-frameworks.md` says:

> Continuous scanning builds requests with no `TargetType` / `TargetNames`. Those scans intentionally inherit the same defaults as an empty `scanV1`.

and there's a test for that shape. Since kubescape.io has a [Continuous scanning](operator/continuous-scanning.md) page, readers there will want to know it's covered too.

By default the value is empty (`[]`): the chart does not write the field, and the operator keeps its legacy fallbacks (see [Scheduled scans](operator/scheduled-scans.md)). Set a list to opt in:

```yaml
defaultFrameworks:
  - allcontrols
  - nsa
  - mitre
```

Examples:

```shell
# AKS CIS benchmark (also scans the "security" framework unless

> # kubescape.triggerSecurityFramework=false)

Reviewer: Wrong value path — this is my error from the last round, sorry. It's `operator.triggerSecurityFramework`, not `kubescape.triggerSecurityFramework`:

```yaml
# helm-charts values.yaml:361,421
operator:
  ...
  triggerSecurityFramework: true
```

and the template reads `.Values.operator.triggerSecurityFramework` (`templates/operator/configmap.yaml:22`). `--set kubescape.triggerSecurityFramework=false` would be silently accepted and do nothing.

```suggestion
# kubescape CIS AKS benchmark (also scans the "security" framework unless
# operator.triggerSecurityFramework=false)
```

(Drop the stray "kubescape" if the suggestion reads oddly — the point is just `operator.` on the second line.)

helm upgrade --install kubescape kubescape/kubescape-operator -n kubescape --create-namespace \
  --set clusterName=`kubectl config current-context` \
  --set 'defaultFrameworks={cis-aks-t1.2.0}'

# All controls plus the current CIS Kubernetes benchmark
helm upgrade --install kubescape kubescape/kubescape-operator -n kubescape --create-namespace \
  --set clusterName=`kubectl config current-context` \
  --set 'defaultFrameworks={allcontrols,cis-v1.10.0}'
```

Clear an existing list with `--set defaultFrameworks=null` (or an empty list in a values file).

Any scan that explicitly sets `targetNames` — including API-triggered scans — uses those targets instead of this list. To override frameworks for the scheduled CronJob only, see [Scheduled scans](operator/scheduled-scans.md).

### Configuring parameters

> See [the GitHub repository for the Kubescape operator](https://github.com/kubescape/helm-charts/blob/main/charts/kubescape-operator/README.md#chart-support) to learn the full set of configuration parameters (including `defaultFrameworks` once that chart release is published).

Reviewer: Small stale hedge: this link points at `blob/main/...README.md`, and `main` **already** has the `defaultFrameworks` row (added by #890, now merged). "once that chart release is published" reads as if it's still missing.

```suggestion
See [the GitHub repository for the Kubescape operator](https://github.com/kubescape/helm-charts/blob/main/charts/kubescape-operator/README.md#chart-support) to learn the full set of configuration parameters, including `defaultFrameworks`.
```
